assets/js/modules/search-console/components/dashboard/DashboardImpressionsWidget.js

```diff
@@ -0,0 +1,107 @@
+/**
+ * DashboardImpressionsWidget component.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { __, _x } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { STORE_NAME } from '../../datastore/constants';
+import { STORE_NAME as CORE_SITE } from '../../../../googlesitekit/datastore/site/constants';
+import { STORE_NAME as CORE_USER } from '../../../../googlesitekit/datastore/user/constants';
+import { extractSearchConsoleDashboardData } from '../../util';
+import { extractForSparkline, getSiteKitAdminURL } from '../../../../util';
+import { trackEvent } from '../../../../util/tracking';
+import whenActive from '../../../../util/when-active';
+import DataBlock from '../../../../components/data-block';
+import Sparkline from '../../../../components/sparkline';
+import PreviewBlock from '../../../../components/preview-block';
+import CTA from '../../../../components/notifications/cta';
+const { useSelect } = Data;
+
+function DashboardImpressionsWidget() {
+	const { data, error } = useSelect( ( select ) => {
+		const store = select( STORE_NAME );
+		const args = {
+			dimensions: 'date',
+			compareDateRanges: true,
+			dateRange: select( CORE_USER ).getDateRange(),
+		};
+
+		const url = select( CORE_SITE ).getCurrentEntityURL();
+		if ( url ) {
+			args.url = url;
+		}
+
+		return {
+			data: store.getReport( args ),
+			error: store.getErrorForSelector( 'getReport', [ args ] ),
+		};
+	} );
+
+	if ( error ) {
+		trackEvent( 'plugin_setup', 'search_console_error', error.message );
+		return <CTA title={ __( 'Something went wrong', 'google-site-kit' ) } description={ error.message } error />;
+	}
+
+	if ( ! data ) {
+		return <PreviewBlock width="100%" height="202px" />;
+	}
+
+	if ( ! data.length ) {
+		return (
+			<CTA
+				title={ __( 'Search Console Data Empty', 'google-site-kit' ) }
+				description={ __( 'Search Console data is not yet available, please check back later.', 'google-site-kit' ) }
+				ctaLink={ '' }
+				ctaLabel={ '' }
+			/>
+		);
+	}
```

Comment:
The same changes requested for `DashboardClicksWidget` should be applied here too.